### DISCUSSION TITLE
Livewire dispatch runs after table refresh

### DISCUSSION THREAD

**Discussion by japgroevemaker** [filamentphp/filament#11614] [Help]
### Package

Table builder

### Package Version

3.0-stable

### How can we help you?

Hello,

I'm having trouble refreshing a table in filament.
What i'm trying to do:

On my dashboard page i have build a from/to date filter which needs to refresh a table with data when i change the from or to date. 

The datePicker is build like this:
```php
public function mount()
    {
        $this->from = Carbon::parse('Now -7 days');
        $this->form->fill();
    }
 
    public function form(Form $form): Form
    {
        return $form
            
            ->schema([
                Grid::make()
                    ->schema([
                        DatePicker::make('from')
                                ->live() 
                                ->native(false)
                                ->displayFormat('Y-m-d')
                                ->minDate(Carbon::parse('Now -90 days'))
                                ->maxDate(Carbon::parse('Now -1 day'))
                                ->default(Carbon::parse('Now -7 days'))
                                ->afterStateUpdated(fn (?string $state) => $this->dispatch('updateFromDate', from: $state)),
            ]);
    }
```

This dispatches an event which is being listened to in my table class like this:
```php
 #[On('updateFromDate')] 
    public function updateFromDate(string $from): void
    {
        dump('The update on dispatch function')
        // things in this function;
       
    }

 public function table(Table $table): Table
 {
dump('The table function')

// and everything a table needs
}
```
However when i change the from date and look at the dumps, the table function is being executed before the updateFromDate function. 

![image](https://github.com/filamentphp/filament/assets/22168803/beb03937-71d2-4f6b-966f-41880e35684f)

Is there a way to execute the updateFromDate function **before** the table function runs?

Thanks in advance

**Comment by mahmoud-757** [comment]
did you find a solution?

**Reply by Bobster94** [reply]
I work together with JapGroevenmaker.
We dit not find a solution yet :(

**Comment by leandrocfe** [comment]
Why not [displaying-filters-above-the-table-content](https://filamentphp.com/docs/3.x/tables/filters/layout#displaying-filters-above-the-table-content) ?

**Reply by Bobster94** [reply]
I work together with JapGroevenmaker.
We have one filter for 3 widgets (see picture) the left widget is a plain livewire component. the right one is a filament chart. both works fine.
But the table is always first render the Table() before the function that is dispatched. so if we set the $this->fromFilter in the dispatch the table is first rendered with the previous $this->fromFilter. 
![Screenshot 2024-03-04 at 12 39 22](https://github.com/filamentphp/filament/assets/10867952/e96b3e6d-7926-43b1-8f56-e1f142cd6c31)
